### ISSUE TITLE
Trouble following packaging libraries tutorial

### ISSUE THREAD

**Issue by JacobLee23** [pypa/packaging-problems#439]
Thank you for providing feedback on Python packaging!

To help us help you, please fill out as much of the following as you can. If a question is not relevant, feel free to skip it.

1. What is your operating system and version?

Windows 10

2. What is your Python version?

Python 3.8.6

3. What version of pip do you have?

pip 21.0.1

4. If following an online tutorial or guide, please provide a link to the page or section giving you trouble:

https://packaging.python.org/tutorials/packaging-projects/

5. Could you describe your issue in as much detail as possible?

```
$ python3 -m build
bash: /c/Users/User/AppData/Local/Microsoft/WindowsApps/python3: Permission denied

$ python -m build
C:\Users\User\AppData\Local\Programs\Python\Python38\python.exe: No module named build.__main__; 'build' is a package and cannot be directly executed
```

**Comment by merwok** [comment]
The first error is strange to me (I don’t use windows), it’s weird that a program exists but can’t be run.

The doc may not be appropriate for windows, if Python 3 is installed as `python`.
(And I thought using the `py` launcher was recommended there.)

To solve your immediate problem: the error you get from `python -m build` tells you that the `python-build` tool is not installed for that Python install.  Run this command first: `python -m pip install build`

**Comment by henryiii** [comment]
You have a file called `build.py` in your directory. Python will use that instead of your package. Please rename or remove the build.py file or otherwise nothing run from that directory will be able to use `build` the python module.

**Comment by merwok** [comment]
From the error message, I think it’s more likely that there is a `build` directory in the current directory.  Unlucky interaction of namespace packages and poorly-chosen name for python-build.

**Comment by henryiii** [comment]
A build directory is generated by pip, and will not contain an `__init__.py`, so if there is a build dir with Python in it, there will be much bigger surprises down the road...

**Comment by layday** [comment]
`build` has a console script, `pyproject-build`, but yes, this is rather unfortunate.

> Pretty sure it never counts as a namespace package.

It does, it just has lower precedence.

**Comment by JLpython-py** [comment]
Yes, so I do have a directory `build/` in my repository, which is under `docs/` (auto-generated by `sphinx`). What do you all recommend I do about this?

**Comment by layday** [comment]
You could:

* Not invoke `build` from the `docs` folder if that isn't your project root
* Use the console script if the it's on your `$PATH`: `pyproject-build`
* Run Python in isolated mode: `python -Im build`

**Comment by henryiii** [comment]
You could also run `python -c "import build; print(build.__path__)"` to see what it's trying to import. :) (`__path__` or `__file__`, the former is needed for namespace packages)

**Comment by xsc27** [comment]
On MacOS 11.6 

```shell
❯ python3 -c "import build; print(build.__path__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'build'

❯ python3 --version
Python 3.9.7
```

😕 

My current workaround is to use `pip`.

```shell
pip3 wheel .
```

**Comment by henryiii** [comment]
Is build installed in your venv?

**Comment by xsc27** [comment]
Oh I see, I missed, https://packaging.python.org/guides/distributing-packages-using-setuptools/#packaging-your-project

https://github.com/pypa/build is new to me.
> A simple, correct PEP517 package builder.

🤔 So `pip` has its own implementation of PEP 517?

Tangentially, I just noticed for the first time `pip` does not give me wheels with the same sha 😖 .

```shell
❯ for i in {1..2}; do pip3 wheel --use-pep517 --use-feature=in-tree-build --no-deps . | grep -oE 'sha256=.*$'; done
sha256=e6f4eb59be3414d860778c195ccd9b802c9646d660d0bb69414cf04cb862a638
sha256=faf7e7f4ba9bd4bd45e57011113569dfac0a24d5f3b34761b818f7b2942037e9
```

`build` gave me reproducible builds. 😀
